Logging features for during and post run

Synicix opened the pull request:
Added:
- get_logs() and get_logs_blocking() that query the logs from the docker API
- Add status Undefined for status to deal with docker api possible return status of restarting, removing, paused, and dead
- Change basic_test to be only asserts, and added excuter_wrapper to handle fixture scaffolding
- Fix basic test assert to list by pod_run as discussed on Friday to avoid collisions
- Add a basic test for log functionality.
- Log are save as a log.txt when saved to disk, and is read back into memory upon loading
- Added hash updating to the logging test.

guzman-raphael reviewed (changes requested):
@Synicix 

- Seems like this is an independent feature. Could you remove the dependency on #72?
- It is missing saving the logs to a separate, single file when storing. We discussed that during YAML serialization, the logs fields should specify the hard-coded, relative path of the log file (e.g. output.log) and it should be skipped when hashing. If you depend on #90, then you can skip hashing in `to_yaml(..)`.

Synicix commented:
@guzman-raphael 
Ah forgot about that feature we discussed. I can quickly added it today.

As for the dependencies, are we able to leave it as that since I expect the 72 to be merge first?

Synicix commented:
@​guzman-raphael

You're right—it doesn't address any major bugs, but it's been a pull request that's been sitting for a few months now, so I thought it made sense to go ahead and merge it.

Also, feel free to correct me if I'm mistaken, but I interpreted your response on pull request #72 as indicating that you'd already reviewed it and were just suggesting that the logging changes be moved to a separate PR. It's possible I misunderstood what you meant.

Synicix commented:
Added
- Pod result save logs to log.txt now, and loads from it when calling from store.

Synicix commented:
For eaiser review in the mean time we decide on whether to remove #72 dependencies or not:
https://github.com/Synicix/orcapod/compare/failure_modes...Synicix:orcapod:logging

Synicix commented on src/uniffi/orchestrator/mod.rs:
```diff
@@ -17,12 +17,16 @@ pub enum ImageKind {
     Completed,
     /// Run failed with the provided error code.
     Failed(i16),
+    /// Catch all for all undefine behavior
```
Resolved

Synicix commented on tests/orchestrator.rs:
```diff
@@ -32,6 +44,7 @@ where
         orchestrator
             .list_blocking()?
             .iter()
+            .filter(|run| **run == pod_run)
```
AI going crazy

Synicix committed:
Save logging patch

Synicix committed:
Readded logging and fix merge issues

Synicix commented:
@guzman-raphael Removed dependencies on #72 

Updated the initial commit with a summary of everything I changed.
Should be much cleaner to merge now.

Synicix commented on src/uniffi/orchestrator/docker.rs:
```diff
@@ -209,8 +235,64 @@ impl Orchestrator for LocalDockerOrchestrator {
+                container_name: &pod_run.assigned_name,
+            })?;
+
+        if !error.is_empty() {
```
Already handle by previous line at 271

Synicix committed:
Update logs test to update the hash also

Synicix commented:
Close do to too many merge conflicts
